loader/loader.py

```diff
@@ -3,56 +3,53 @@
 "http://ghtorrent-downloads.ewi.tudelft.nl/mongo-daily/" and store them into target directory.
 """
 import argparse
+from datetime import datetime
+from functools import partial
+from multiprocessing import Pool
 import os
+import re
 import tarfile
-from typing import List, Callable, Optional
+from typing import List
 
 from bs4 import BeautifulSoup
+import requests
 from tqdm import tqdm
-import urllib.request
 
 
-def download_progress_hook(p_bar: tqdm) -> Callable[[int, int, Optional[int]], None]:
-    """
-    Wraps tqdm instance
-    :param p_bar: tqdm instance responible for showing download progress
-    :return: function update_to which updates tqdm state
-    """
-    last_block = [0]
-
-    def update_to(block_num: int = 1, block_size: int = 1,
-                  total_size: Optional[int] = None) -> None:
-        """
-        Update tqdm state
-        :param block_num: Number of blocks transferred so far
-        :param block_size: Size of each block
-        :param total_size: Total size (in tqdm units)
-        :return: None
-        """
-        if total_size not in (None, -1):
-            p_bar.total = total_size
-        p_bar.update((block_num - last_block[0]) * block_size)
-        last_block[0] = block_num
-
-    return update_to
-
-
-def extract_archive_links(url: str) -> List[str]:
+def extract_archive_links(url: str, start_date: str, end_date: str) -> List[str]:
     """
     Extract tar files links from GHTorrent html page to list
     :param url: url of page with daily dumps tars
+    :param start_date: Start date in ISO format
+    :param end_date: End date in ISO format
     :return: list of tar files links
     """
-    html_page = urllib.request.urlopen(url)
+    html_page = requests.get(url).text
     soup = BeautifulSoup(html_page, features="html.parser")
     tar_files_links = []
     for link in soup.findAll("a", href=True):
         filename = link.get("href")
-        if filename.endswith("tar.gz"):
+        if filename.endswith("tar.gz") and is_between_dates(filename, start_date, end_date):
             tar_files_links.append(url + filename)
     return tar_files_links
 
 
+def is_between_dates(tar_filename: str, start_date: str, end_date: str) -> bool:
+    """
+    Check that current dump date is between start_date and end_date
+    :param tar_filename: Tar filename which contains daily dump date
+    :param start_date: Start date in ISO format
+    :param end_date: End date in ISO format
+    :return: True if tar file date is between start date and end date, else - False
+    """
+    iso_format = "%Y-%m-%d"
+    start_date = datetime.strptime(start_date, iso_format)
+    end_date = datetime.strptime(end_date, iso_format)
+    cur_date = re.search("mongo-dump-(.*).tar.gz", tar_filename).group(1)
```

Comment:
```suggestion
    tar_date = re.search("mongo-dump-(.*).tar.gz", tar_filename).group(1)
```
It's not current date - let's use tar or archive date